packages/amplify_auth_plugin_interface/lib/src/Session/AuthUserAttributeOptions.dart

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+class AuthUserAttributeOptions {
+
+  const AuthUserAttributeOptions();
+  Map<String, dynamic> serializeAsMap() {
+    throw UnimplementedError('serializeAsMap() has not been implemented on AuthUserAttributeOptions.');
+  }
+}
```

Comment:
What's the point of this empty class and part of a request signature when no data can be provided? Seems confusing.

Reply:
This is in anticipation of future, non-Cognito plugins which may have options. I believe that native libraries do this as well for this API.